File: c/insertion.c
#include <stdio.h>
#include <stdlib.h>

/*
/**
 * This will swap two pointers around
 * @param c Pointer to the first data to be swapped
 * @param d Pointer to the second data to be swapped
 *//*
void swap(int *c, int *d)
{
    int temp = *c;
    *c = *d;
    *d = temp;
}
*/

/**
 * This will sort data using insertion sort.  Done in ascending order
 * @param data Pointer to the list to be sorted
 */
int insertion(Tuple *data)
{
    int i,j,t;

    for (i=1; i<data->sizeOfList; i++)
    {
        j = i;
        t = *((data-> listToUse) + j);

        while (j > 0 && t < *((data->listToUse)+ (j-1)))
        {
            *((data->listToUse) + j) = *((data->listToUse) + (j-1));
            j = j -1;
        }

        *((data->listToUse) + j) = t; //Puts the data in the correct position
    }

    return 0;
}

File: python/insertion.py
def insertion(data_array):

    """ This will sort the data numerically in ascending order
    data_array - This is the data to be sorted
    returns the data to be sorted
    """
    
    for index in range(1, len(data_array)): #Will insert each data in the right position

        current_data = data_array[index]  #Gets the current data
        position = index #Gets the index
        
        while position > 0 and data_array[position - 1] > current_data: #Whilst the index is larger than 0 and the data to be inserted is smaller than the data in the sorted part of the list
            
            data_array[position] = data_array[position - 1]
            position = position - 1
        
        data_array[position] = current_data
        
    return data_array
